### PR TITLE
Docs: Initial sequence of pages in Documentation

### PR DESCRIPTION
**Pull request by AdarshRawat1**
## What 
Reordered the sequence of topics in generated documentation. _View mockup for Changes [live here](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/)_ 

#### Removed following outdated pages from documentation, until they are updated.
- https://github.com/p4lang/p4c/blob/main/backends/p4tools/cmake/README.md
- https://github.com/p4lang/p4c/blob/main/backends/p4tools/common/lib/README.md
- https://github.com/p4lang/p4c/blob/main/backends/p4tools/common/README.md
- https://github.com/p4lang/p4c/blob/main/backends/p4tools/modules/testgen/lib/README.md

## Why 
Updated sequence of files listed by Doxygen output based on discussion in
- #4746  

#### Current Sequence 
| List |
|--------|
| [P4C](https://p4lang.github.io/p4c/index.html)
[Repository](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2docs_2_r_e_a_d_m_e.html)
[Revision History](https://p4lang.github.io/p4c/md_00__revision__history.html)
[Overview](https://p4lang.github.io/p4c/md_01__overview.html)
[P4 Compiler Intermediate representation](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2docs_2_i_r.html)
[Behavioral Model Backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2bmv2_2_r_e_a_d_m_e.html)
[DPDK backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2dpdk_2_r_e_a_d_m_e.html)
[PSA implementation for eBPF backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2ebpf_2psa_2_r_e_a_d_m_e.html)
[eBPF Backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2ebpf_2_r_e_a_d_m_e.html)
[Graphs Backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2graphs_2_r_e_a_d_m_e.html)
[p4fmt (P4 Formatter)](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4fmt_2_r_e_a_d_m_e.html)
[P4test Backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4test_2_r_e_a_d_m_e.html)
[P4Tools CMake](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2cmake_2_r_e_a_d_m_e.html)
[P4Tools Common Library Code](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2common_2lib_2_r_e_a_d_m_e.html)
[P4Tools Common C++ code](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2common_2_r_e_a_d_m_e.html)
[P4Tools Contributors](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2_c_o_n_t_r_i_b_u_t_o_r_s.html)
[README](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2modules_2smith_2_r_e_a_d_m_e.html)
[P4Testgen Benchmarks](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2modules_2testgen_2benchmarks_2_r_e_a_d_m_e.html)
[P4Testgen Library](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2modules_2testgen_2lib_2_r_e_a_d_m_e.html)
[P4Testgen](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2modules_2testgen_2_r_e_a_d_m_e.html)
[P4Testgen BMv2 target tests](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2modules_2testgen_2targets_2bmv2_2test_2_r_e_a_d_m_e.html)
[P4Tools - Testing Tools For P4 Targets](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2p4tools_2_r_e_a_d_m_e.html)
[TC backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2tc_2_r_e_a_d_m_e.html)
[uBPF Backend test programs](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2ubpf_2docs_2_e_x_a_m_p_l_e_s.html)
[Introduction to uBPF Backend](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2ubpf_2_r_e_a_d_m_e.html)
[uBPF Backend testing](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2ubpf_2tests_2_r_e_a_d_m_e.html)
[Common P4C utility functions](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2lib_2_r_e_a_d_m_e.html)
[P4C Intermediate Representation (IR) Classes](https://p4lang.github.io/p4c/irdefs.html)
[Changelog](https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2_c_h_a_n_g_e_l_o_g.html)
[Deprecated List](https://p4lang.github.io/p4c/deprecated.html)
[Todo List](https://p4lang.github.io/p4c/todo.html)
[Topics](https://p4lang.github.io/p4c/topics.html)
[Namespaces](https://p4lang.github.io/p4c/namespaces.html)
[Classes](https://p4lang.github.io/p4c/annotated.html) | 

#### Updated Sequence  [LIVE](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/) 
| List |
|--------|
|[P4C](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/index.html)
[Repository](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2docs_2_r_e_a_d_m_e.html)
[Revision History](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md_00__revision__history.html)
[Overview](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md_01__overview.html)
[P4 Compiler Intermediate representation](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2docs_2_i_r.html)
[Common P4C utility functions](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2lib_2_r_e_a_d_m_e.html)
[P4C Intermediate Representation (IR) Classes](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/irdefs.html)
[Changelog](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2_c_h_a_n_g_e_l_o_g.html)
[Behavioral Model Backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2bmv2_2_r_e_a_d_m_e.html)
[DPDK backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2dpdk_2_r_e_a_d_m_e.html)
[eBPF Backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2ebpf_2_r_e_a_d_m_e.html)
[PSA implementation for eBPF backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2ebpf_2psa_2_r_e_a_d_m_e.html)
[Graphs Backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2graphs_2_r_e_a_d_m_e.html)
[p4fmt (P4 Formatter)](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4fmt_2_r_e_a_d_m_e.html)
[P4test Backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4test_2_r_e_a_d_m_e.html)
[P4Tools - Testing Tools For P4 Targets](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4tools_2_r_e_a_d_m_e.html)
[P4Tools Contributors](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4tools_2_c_o_n_t_r_i_b_u_t_o_r_s.html)
[README](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4tools_2modules_2smith_2_r_e_a_d_m_e.html)
[P4Testgen](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4tools_2modules_2testgen_2_r_e_a_d_m_e.html)
[P4Testgen Benchmarks](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4tools_2modules_2testgen_2benchmarks_2_r_e_a_d_m_e.html)
[P4Testgen BMv2 target tests](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2p4tools_2modules_2testgen_2targets_2bmv2_2test_2_r_e_a_d_m_e.html)
[TC backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2tc_2_r_e_a_d_m_e.html)
[Introduction to uBPF Backend](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2ubpf_2_r_e_a_d_m_e.html)
[uBPF Backend test programs](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2ubpf_2docs_2_e_x_a_m_p_l_e_s.html)
[uBPF Backend testing](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/md__b_1_2_o_s_p_2p4c_2backends_2ubpf_2tests_2_r_e_a_d_m_e.html)
[Deprecated List](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/deprecated.html)
[Todo List](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/todo.html)
[Topics](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/topics.html)
[Namespaces](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/namespaces.html)
[Classes](https://6679d5c80692ef2a416f96fb--bespoke-douhua-c67c10.netlify.app/annotated.html)|

### CODE REVIEW

File: docs/doxygen/doxygen.cfg [hunk 1]
```diff
@@ -1008,10 +1028,13 @@ RECURSIVE              = YES
 
 EXCLUDE                = ../../frontends/p4/README.md \
                          ../../midend/README.md \
-                         ../../extensions/p4_tests \
-                         ../../control-plane/PI \
                          ./awesome_css \
-                         ./build 
+                         ./build \
+                         ../../backends/p4tools/cmake/README.md \
+                         ../../backends/p4tools/cmake/README.md \
```
Comment (jafingerhut) [inline docs/doxygen/doxygen.cfg:1034]: Looks like an unnecessary duplicate of the line above?

**Review by jafingerhut — APPROVED** [review]
LGTM

**Comment by jafingerhut** [comment]
@fruffy It appears DCO is required in order to click the Merge button on Github.  Note: That sounds GOOD to me.  Just wanted to let you know, since you mentioned recently that you thought it was optional right now.

@AdarshRawat1 I think the only thing needed to commit this is for you to add DCO.  You can can click the "Details" link to the right of the failing check labeled "DCO" for instructions on how.

**Comment by fruffy** [comment]
> @fruffy It appears DCO is required in order to click the Merge button on Github. Note: That sounds GOOD to me. Just wanted to let you know, since you mentioned recently that you thought it was optional right now.

I think this is because this is a merge pull request.

**Comment by AdarshRawat1** [comment]
> @AdarshRawat1 I think the only thing needed to commit this is for you to add DCO. You can can click the "Details" link to the right of the failing check labeled "DCO" for instructions on how.

Thanks, Done !!

**Comment by jafingerhut** [comment]
I was wrong.  You also need to change the PR so it is no longer in Draft status.

**Comment by fruffy** [comment]
Seems like this PR has broken CI, I would use rebase merge instead of conventional merging. It works much better on Github.

**Review by fruffy — APPROVED** [review]

**Pull request merged**